autonomy/cli/scaffold_fsm.py

```diff
@@ -495,6 +496,96 @@ def _get_round_behaviour_section(self) -> str:
         )
 
 
+class PayloadsFileGenerator(AbstractFileGenerator):
+    """File generator for 'payloads.py' modules."""
+
+    FILENAME = PAYLOADS_FILENAME
+
+    PAYLOADS_FILE = dedent(
+        """\
+        \"\"\"This module contains the transaction payloads of the {FSMName}.\"\"\"
+
+        from enum import Enum
+        from typing import Any, Dict, Hashable, Optional
+
+        from packages.valory.skills.abstract_round_abci.base import BaseTxPayload
+
+
+        class TransactionType(Enum):
+            \"\"\"Enumeration of transaction types.\"\"\"
+
+            # TODO: define transaction types: e.g. TX_HASH: "tx_hash"
+            ...
+
+            def __str__(self) -> str:
+                \"\"\"Get the string value of the transaction type.\"\"\"
+                return self.value
+
+        """
+    )
+
+    BASE_PAYLOAD_CLS = dedent(
+        """\
+        class Base{FSMName}Payload(BaseTxPayload):
+            \"\"\"Base payload for {FSMName}.\"\"\"
+
+            def __init__(self, sender: str, content: Hashable, **kwargs: Any) -> None:
```

Comment:
This only supports a single content. I'd prefer we do this refactoring separately. As this is a largely non-consequential PR we can leave it now, but this class makes not much sense to me.

Reply:
would be a pretty good rule to have payloads with a single piece of content. Although we should make the `Hashable` into `ProtobufSerializable` and then it can be a nested structure as well